### PR TITLE
feat: Add search history & top searches

### PR DESCRIPTION
**Pull request by filipewl**
## How does it work?

See https://github.com/vtex-sites/gatsby.store/pull/67 for more details.

## How to test it?

Because the intelligent search's top search data comes from [real usage data from the last 14 days](https://help.vtex.com/en/tracks/vtex-intelligent-search--19wrbB7nEQcmwzDPl1l4Cb/4gXFsEWjF7QF7UtI2GAvhL#:~:text=Section%20information%20Most%20searched%20terms%20and%20Search%20suggestions%20are%20updated%20daily%20using%20the%20data%20from%20the%20last%2014%20days%20of%20the%20customers%27%20interaction%20with%20the%20search.), as of today, we don't have any top search data to show for the **storeframework** store ID. That means the preview deploy won't display the Top Search component with data. Because of that, and for testing purposes, I've created a temporary branch where I switched the store ID for **storecomponents**, it has top search data and you can preview it working in [this preview deploy](https://sfj-0256cc7--nextjs.preview.vtex.app/).

NextJS Search History n Top Search

## Known Issues

- The [Storybook's `TopSearch` storie](https://nextjs-store-storybook-git-feat-search-history-n-top-faststore.vercel.app/?path=/docs/organisms-search-topsearch--default) is not loading data correctly and will be fixed in a later PR.

### CODE REVIEW

**Review by hellofanny — APPROVED** [review]
Looks good! Well done 🚀 

image

File: src/components/search/Suggestions/suggestions.scss [hunk 1]
```diff
@@ -27,6 +27,14 @@
   }
 }
 
+// Temporarily increased CSS specificity because the normal link blue color
+// was winning. Hoping this will be fixed when using CSS modules.
```
Comment (eduardoformiga) [inline src/components/search/Suggestions/suggestions.scss:31]: should we add as `// TODO`?

**Review by eduardoformiga — APPROVED** [review]
Amazing! ✨ 

I noticed a border-radius issue that is not so important and we can deal with it afterward
29tFeTokGZ

File: src/components/search/Suggestions/suggestions.scss (hunk 1 above)
Comment (filipewl) [reply src/components/search/Suggestions/suggestions.scss:31]: Good idea! [Done](https://github.com/vtex-sites/nextjs.store/pull/112/commits/6300c54e3803176b1302fec2ec4c43ae5fd4627d).

**Comment by hellofanny** [comment]
> Amazing! ✨
> 
> I noticed a border-radius issue that is not so important and we can deal with it afterward 29tFeTokGZ [ ](https://user-images.githubusercontent.com/11325562/173671688-e30b8bef-2ec1-4086-b7ca-bc8d68161509.gif) 29tFeTokGZ

I think this was intentional. The bottom corners should be flat (according to the prototype 🤔 )

**Comment by eduardoformiga** [comment]
> > Amazing! ✨
> > I noticed a border-radius issue that is not so important and we can deal with it afterward 29tFeTokGZ
> >       
> >         [
> >           
> >         ](https://user-images.githubusercontent.com/11325562/173671688-e30b8bef-2ec1-4086-b7ca-bc8d68161509.gif)
> >         
> >             29tFeTokGZ
> >           
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/11325562/173671688-e30b8bef-2ec1-4086-b7ca-bc8d68161509.gif)
> 
> I think this was intentional. The bottom corners should be flat (according to the prototype 🤔 )

Even when there are no results for the search term?

**Comment by filipewl** [comment]
@eduardoformiga and @hellofanny, I've created an [internal task](https://vtex-dev.atlassian.net/browse/FSSS-419) to deal with the border radius issue because I couldn't think of a very straightforward solution. Thanks for all the feedback! 
🙇

**Pull request merged**